rust/processor/migrations/2024-05-22-200847_add_v1_migration_views/up.sql

```diff
@@ -0,0 +1,319 @@
+
+-- replace `current_coin_balances` with `current_fungible_asset_balances`
+CREATE OR REPLACE VIEW legacy_migration_v1.current_coin_balances AS
+    SELECT
+        owner_address,
+        substring(asset_type, 64) as coin_type_hash,
+        asset_type as coin_type,
+        amount,
+        transaction_version as last_transaction_version,
+        transaction_timestamp as last_transaction_timestamp,
+        inserted_at
+    FROM public.current_fungible_asset_balances
+    WHERE token_standard = 'v1'
+    ;
+CREATE INDEX lm1_ccb_ct_a_index ON public.current_fungible_asset_balances USING btree (asset_type, amount)
+
+-- replace `token_activities` with `token_activities_v2`
+-- token_activities_v2.token_data_id is 0x prefixed, but token_activities.token_data_id is not. We need to create an index on the substring
+CREATE OR REPLACE VIEW legacy_migration_v1.token_activities AS
+    SELECT
+        tav.transaction_version,
+        event_account_address,
+        -- These were only used for hashing pk in v1 table
+        0 as event_creation_number,
+        0 as event_sequence_number,
+        tdv.collection_id  as collection_data_id_hash,
+        ltrim(tav.token_data_id, '0x') as token_data_id_hash,
+        property_version_v1 AS property_version,
+        cv.creator_address,
+        cv.collection_name,
+        tdv.token_name AS "name",
+        "type" AS transfer_type,
+        from_address,
+        to_address,
+        token_amount,
+        -- These are not columns in v2
+        NULL AS coin_type,
+        NULL AS coin_amount,
+        tav.inserted_at,
+        tav.transaction_timestamp,
+        event_index
+    FROM public.token_activities_v2 tav 
+    JOIN token_datas_v2 tdv 
+    ON tav.token_data_id = tdv.token_data_id AND tav.transaction_version = tdv.transaction_version
+    JOIN collections_v2 cv
+    ON tdv.collection_id = cv.collection_id AND tdv.transaction_version = cv.transaction_version
+    WHERE tav.token_standard = 'v1'
+;
+-- token_activities_v2 already has index on token_data_id
+CREATE INDEX lm1_tdv_tdi_tv_index ON public.token_datas_v2 USING btree (token_data_id, transaction_version)
+CREATE INDEX lm1_cv_ci_tv_index ON public.collections_v2 USING btree (collection_id, transaction_version)
+CREATE INDEX lm1_ta_tdih_pv_index ON public.token_activities_v2 USING btree (token_data_id, property_version_v1)
+
+-- replace `token_ownerships` with `token_ownerships_v2`
+CREATE OR REPLACE VIEW legacy_migration_v1.token_ownerships AS 
+    SELECT
+        tov.token_data_id AS token_data_id_hash,
+        property_version_v1 AS property_version,
+        tov.transaction_version,
+        -- this is a bit concerning
+        '' AS table_handle,
+        creator_address,
+        collection_name,
+        tdv.token_name AS name,
+        owner_address,
+        amount,
+        table_type_v1 AS table_type,
+        tov.inserted_at,
+        tdv.collection_id AS collection_data_id_hash,
```

Comment:
values are different

Reply:
can you give an example?

Reply:
similar to coins, logic would be `encode(
sha256(
(collection_creator_address || '::' || token_name)::bytea
), 'hex')`

from here https://github.com/aptos-labs/aptos-indexer-processors/blob/5244b84fa5ed872e5280dc8df032d744d62ad29d/python/utils/token_utils.py#L44

Reply:
The processing code does the hash already - https://github.com/aptos-labs/aptos-indexer-processors/blob/48d779449caa5b10a51a6319c61992af0edb52ce/rust/processor/src/models/token_v2_models/v2_collections.rs#L264

Reply:
just realized that, so just need `ltrim`